client/transaction-pool/src/revalidation.rs

```diff
@@ -68,13 +68,20 @@ async fn batch_revalidate<Api: ChainApi>(
 	let mut invalid_hashes = Vec::new();
 	let mut revalidated = HashMap::new();
 
-	for ext_hash in batch {
-		let ext = match pool.validated_pool().ready_by_hash(&ext_hash) {
-			Some(ext) => ext,
-			None => continue,
-		};
+	let validation_results = futures::future::join_all(
+		batch.into_iter().filter_map(|ext_hash| {
+			pool.validated_pool().ready_by_hash(&ext_hash).map(|ext| {
+				let api = api.clone();
+				async move {
+					api.validate_transaction(&BlockId::Number(at), ext.source, ext.data.clone())
+						.map(|validation_result| (validation_result, ext_hash.clone(), ext)).await
+				}
```

Comment:
```suggestion
				api.validate_transaction(&BlockId::Number(at), ext.source, ext.data.clone())
					.map(|validation_result| (validation_result, ext_hash.clone(), ext))
```

Should work as well?

Reply:
No, I think I tried, api needs to be clone in every parallel future I think